pkg/leader/types.go

```diff
@@ -0,0 +1,204 @@
+type Review struct {
+	Author   Author `json:"author"`
+	Comments struct {
+		TotalCount int `json:"totalCount"`
+	} `json:"comments"`
+}
+
+// Label represents GitHub label as used on Issues and PRs.
+type Label struct {
+	Name string `json:"name"`
+}
+
+// Author represents GitHub user who authored PR, Review, etc.
+type Author struct {
+	Login     string `json:"login"`
+	URL       string `json:"url"`
+	AvatarURL string `json:"avatarURL"`
+}
+
+// ChartData holds all aggregate PR, review and comment data relevant to
+// visualisation.
+type ChartData struct {
+	Authors         map[string]Author `json:"authors"` // keyed by author.login (github username)
+	Charts          []Chart           `json:"chart"`
+	BotCommentCount int               `json:"botComments"`
+	Repository      Repository        `json:"repository"`
+}
+
+// Chart contains data points, aggregated and meta data for charting.
+type Chart struct {
+	Title      string  `json:"title"`
+	MaxCount   int     `json:"maxCount"`
+	TotalCount int     `json:"totalCount"`
+	Points     []Point `json:"points"`
+}
+
+// Point contains chartable data point per for a GitHub user. Count may
+// represent total number of Pull Requests merged, merged or open, reviews,
+// comments etc. per user.
+type Point struct {
+	Author string `json:"author"`
+	Count  int    `json:"count"`
+}
+
+// Repository meta data used in ChartData
+type Repository struct {
+	Name  string `json:"name"`
+	Owner string `json:"owner"`
+	URL   string `json:"url"`
+}
+
+type countByAuthor struct {
+	pr      map[string]int
+	open    map[string]int
+	review  map[string]int
+	comment map[string]int
+}
+
+func newCountByAuthor() countByAuthor {
+	return countByAuthor{
+		pr:      map[string]int{},
+		open:    map[string]int{},
+		review:  map[string]int{},
+		comment: map[string]int{},
+	}
+}
+
+// ChartDataFromPRs creates relevant struct for visualising aggregated
+// PR, review and comment data. Input to this function is PR data as
+// returned by GitHub API v4 (GraphQL).
+func ChartDataFromPRs(gqlPRs []PRNode, config ChartDataConfig) ChartData {
+	countByAuthor := newCountByAuthor()
+	authors := map[string]Author{}
+	for _, pr := range gqlPRs {
+		if !pr.CreatedAt.After(config.CreatedAfter) {
+			continue
+		}
+		if !hasLabLabel(pr.Lables.LabelNodes, config.LabelRegexp) {
+			continue
+		}
+		author := pr.Author.Login
+		authors[author] = pr.Author
+		if pr.State == "MERGED" {
+			countByAuthor.pr[author]++
+		}
+		if pr.State == "MERGED" || pr.State == "OPEN" {
+			countByAuthor.open[author]++
+		}
+		for _, review := range pr.Reviews.ReviewNodes {
+			author := review.Author.Login
+			countByAuthor.review[author]++
+			// Add 1 to comment count because the review itself must contain a "comment" which isn't counted.
+			countByAuthor.comment[author] = countByAuthor.comment[author] + 1 + review.Comments.TotalCount
+			authors[review.Author.Login] = review.Author
```

Comment:
```
authors[author] = review.Author
```
and maybe move it to the second line of the loop so it looks the same as the PR author stuff above

Reply:
Done.